tests/test_sshfsmount.cpp

```diff
@@ -80,9 +80,11 @@ struct SshfsMount : public mp::test::SftpServerTest
             if (next_expected_cmd != commands.end())
             {
                 // Check if the first of the remaining commands is being executed.
-                if (cmd == next_expected_cmd->first)
+                if (cmd.find(next_expected_cmd->first) != std::string::npos)
                 {
-                    output = next_expected_cmd->second;
+                    // We add a character to the output because the function gathering the output of the executed
+                    // commands uses it to mark the end of the output.
+                    output = std::string{next_expected_cmd->second + '-'};
                     remaining = output.size();
                     ++next_expected_cmd;
                     invoked = true;
```

Comment:
I don't like this, we really want to test for equality. One more reason against the `echo` approach IMO.

The test now assumes the command is executed via `run_string_cmd`. What if one of the calls goes through `run_cmd` directly? And if it's not meant to, why have `run_cmd` and `run_string_cmd` separate?

Reply:
Note I introduced the == in a previous PR, but now the `.find()` is needed. I never liked how the tests looked after introducing `run_string_cmd` :)

After reading _all_ yout comments, I see `run_string_cmd` pollutes the code, and introduces these glitches in the tests.